packages/rum-vue/src/error-handler.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+export function setUpErrorHandler(Vue, apm) {
+  /**
+   * If the user already installed a global error handler
+   * we should call them after capturing it internally
+   */
+  const previousErrorHandler = Vue.config.errorHandler
+
+  return (error, vm, info) => {
+    if (vm && vm.$options) {
+      const options = vm.$options
+      const component = options.name || options._componentTag
+
+      error.component = component || 'anonymous'
+      error.file = options.__file || ''
+    }
+
+    /**
+     * Lifecycle hook the error was thrown if available
+     */
+    if (info) {
+      error.lifecycleHook = info
+    }
+
+    apm.captureError(error)
+
+    if (typeof previousErrorHandler === 'function') {
+      previousErrorHandler.call(this, error, vm, info)
+    }
+    /**
+     * Logs the error to browser console similar
+     * to original Vue logError
+     */
+    console.error(error)
```

Comment:
Is it possible to use the original vue error handler here? similar to what we did with angular.

Reply:
Not really, Its not exported, if this config `errorHandler` is present then it takes the preference. But it does log to the console. 

https://github.com/vuejs/vue/blob/7186940143704acc4ec046132f6a56e9c983e510/src/core/util/error.js#L73-L83

We can additionally do Vue.util.warn but then we would need to check if env is Prod or Dev and it brings in additional reliance on this method. since its a utility.